subsys/net/l2/ethernet/ethernet.c

```diff
@@ -500,7 +500,16 @@ static int ethernet_send(struct net_if *iface, struct net_pkt *pkt)
 	u16_t ptype;
 	int ret;
 
-	if (IS_ENABLED(CONFIG_NET_IPV4) &&
+	if (IS_ENABLED(CONFIG_NET_SOCKETS_RAW) &&
+	    net_pkt_family(pkt) == AF_PACKET) {
+		ret = api->send(net_if_get_device(iface), pkt);
+		if (!ret) {
+			ret = net_pkt_get_len(pkt);
+			net_pkt_unref(pkt);
+		}
+
+		return ret;
```

Comment:
You are not updating eth stats here, 

so don't send here, just a goto that goes to line 571 and that's it